chapters/envs.rst

```diff
@@ -14,157 +16,168 @@ Here is the list of environment variables that ``jina`` respects during runtime.
+    The Pod name set when a Pod started when ``--log-with-pod-name`` is on, this should not given manually by users.
 
-    :default: ``INFO``
+    :default: unset
 
-.. confval:: JINA_LOG_SSE
+.. confval:: JINA_RAISE_ERROR_EARLY
 
-    Turn on the server side event logging for all pea, pods, executors running in the context.
+    Raise exception immedidately instead of passing it forward in the flow. Useful in debugging.
 
     :default: unset
 
-.. confval:: JINA_LOG_PROFILING
+.. confval:: JINA_RANDOM_PORTS
 
-    Turn on the server side event logging for profiling, otherwise you won't able to see "task" tab in the dashboard.
+    Set to any non-empty will trigger the random search for available port.
 
     :default: unset
 
-.. confval:: JINA_LOG_LONG
+.. confval:: JINA_RANDOM_PORT_MAX
 
-    When set, the filename, function name and line number will be displayed as well.
+    Set to '65535' as the max value when searching random port.
 
     :default: unset
 
-.. confval:: JINA_DEFAULT_HOST
+.. confval:: JINA_RANDOM_PORT_MIN
 
-    The default host address of Jina.
+    Set to '49153' as the min value when searching random port.
 
-    :default: `0.0.0.0`
+    :default: unset
 
-.. confval:: JINA_TEST_CONTAINER
+.. confval:: JINA_SOCKET_HWM
 
-    If set, then all container-related tests will be conducted in the unit test.
+    High-watermarks of ZMQ send & receive sockets. Reference: http://api.zeromq.org/master:zmq-setsockopt
 
-    :default: unset
+    :default: 4
 
 .. confval:: JINA_TEST_PRETRAINED
 
     If set, then all pretrained model-related tests will be conducted in the unit test.
 
     :default: unset
 
-.. confval:: JINA_POD_NAME
-
-    The Pod name set when a Pod started when ``--log-with-pod-name`` is on, this should not given manually by users.
-
-    :default: unset
-
 .. confval:: JINA_TEST_GPU
 
     If set, then all gpu-related tests will be conducted in the unit test.
 
     :default: unset
 
-.. confval:: JINA_BINARY_DELIMITER
+.. confval:: JINA_VCS_VERSION
 
-    The delimiter used in :class:`BasePbIndexer`. Often use a delimiter phrase that you can determine does not occur. (Think of the mime message boundaries)
+    Git version of ``jina``. This is used when ``--check-version`` is turned on. For official docker image of ``jina``, ``JINA_VCS_VERSION`` is automatically set to the git version during the building procedure.
 
-.. confval:: JINA_DISABLE_UVLOOP
+    :default: the git head version for ``jina`` image. If you are using ``jina`` locally outside docker container then this is unset.
 
-    ``uvloop`` is ultra fast implementation of the asyncio event loop on top of libuv. Since Jina 0.3.6, Jina relies on ``uvloop`` to manage the sockets and streams.
+.. confval:: JINA_WARN_UNNAMED
+
+    Set to any non-empty value to turn on the warning for unnamed executors.
 
     :default: unset
 
-.. confval:: JINA_RAISE_ERROR_EARLY
+.. confval:: JINA_WORKSPACE
```

Comment:
where is this used?

Reply:
JINA_LOG_WORKSPACE  is used in optimizer/discovery.py and mostly used in tests, examples.

Reply:
if is used by discovery is good, but in tests and examples is not cores concern, here there should only be listed those used in the core

Reply:
Change the description to the one in `discovery`.

Reply:
Just changed